stl/inc/ranges

```diff
@@ -6837,6 +6837,602 @@ namespace ranges {
+                auto _Smallest_abs_difference_value =
+                    _ResultType{_RANGES distance(_STD get<0>(_Lhs_tuple), _STD get<0>(_Rhs_tuple))};
+
+                const auto _Evaluate_single_distance_lambda =
+                    [&_Lhs_tuple, &_Rhs_tuple, &_Smallest_difference_value,
+                        &_Smallest_abs_difference_value]<size_t _CurrIndex>() {
+                        const auto _Curr_difference_value =
+                            _ResultType{_STD get<_CurrIndex>(_Lhs_tuple) - _STD get<_CurrIndex>(_Rhs_tuple)};
+                        const auto _Curr_abs_difference_value = _ResultType{
+                            _RANGES distance(_STD get<_CurrIndex>(_Lhs_tuple), _STD get<_CurrIndex>(_Rhs_tuple))};
+
+                        if (_Curr_abs_difference_value < _Smallest_abs_difference_value) {
+                            _Smallest_difference_value     = _Curr_difference_value;
+                            _Smallest_abs_difference_value = _Curr_abs_difference_value;
+                        }
+                    };
+
+                const auto _Evaluate_all_other_distances_lambda =
+                    [&_Evaluate_single_distance_lambda]<size_t... _Indices>(index_sequence<_Indices...>) {
+                    ((_Evaluate_single_distance_lambda.template operator()<_Indices>()), ...);
+                };
+
+                // Start from index 1, since we already evaluated 0.
+                constexpr auto _Offset_index_sequence = _Zip::_Get_offset_index_sequence<_Tuple_size, 1>();
+                _Evaluate_all_other_distances_lambda(_Offset_index_sequence);
+
+                return _Smallest_difference_value;
+            }
+        }
+
+        template <typename _LHSItrOrSentTuple, typename _RHSItrOrSentTuple>
+            requires(tuple_size_v<decay_t<_LHSItrOrSentTuple>> == tuple_size_v<decay_t<_RHSItrOrSentTuple>>)
+        _NODISCARD constexpr bool _Evaluate_iterator_sentinel_equality(_LHSItrOrSentTuple&& _Lhs_tuple,
+            _RHSItrOrSentTuple&& _Rhs_tuple) noexcept(_Misc_iterator_sentinel_noexcept_solver<bool,
+            decay_t<_LHSItrOrSentTuple>, decay_t<_RHSItrOrSentTuple>>::_Is_evaluate_equality_noexcept) {
+            constexpr auto _Index_sequence = make_index_sequence<tuple_size_v<decay_t<_LHSItrOrSentTuple>>>{};
+            const auto _Evaluate_equality_lambda =
+                [&_Lhs_tuple, &_Rhs_tuple ]<size_t... _Indices>(index_sequence<_Indices...>) {
+                return ((_STD get<_Indices>(_Lhs_tuple) == _STD get<_Indices>(_Rhs_tuple)) || ...);
+            };
+
+            return _Evaluate_equality_lambda(_Index_sequence);
+        }
+
+        class _Zip_function {
+        public:
+            _Zip_function() noexcept = default;
+
+            template <input_range... _ViewTypes>
+            constexpr auto operator()(_ViewTypes&&... _Views) const
+                noexcept(noexcept(views::empty<tuple<>>) && noexcept(zip_view{_STD forward<_ViewTypes>(_Views)...})) {

```

Comment:
Great catch! As it turns out, that's actually *not* okay, because the expression `zip_view{}` is ill-formed. This is because `zip_view` requires the parameter pack to contain at least one view type. I'll be pushing a commit which solves this by using a `consteval` function with `if constexpr` to determine the correct `noexcept` specification.